invenio_communities/communities/permissions.py

```diff
@@ -1,106 +1,88 @@
+        """Filters for current identity as owner."""
+        #TODO do we ever expect more than one user?
+        # users = [n.value for n in identity.provides if n.method == "id"]
+        # return Q(
+        #     "terms",
+        #     **{"access.owned_by.user": users}
+        #     )
+        return Q()
+
+
+class CommunityOwners(Generator):
+    """Allows community owners."""
+
+    def needs(self, record=None, **kwargs):
+        """Enabling Needs."""
+        #TODO check if multiple or one owner
+        return [UserNeed(record.access.owner.owner_id)]
+
+    def query_filter(self, identity=None, **kwargs):
+        """Filters for current identity as owner."""
+        #TODO do we ever expect more than one user?
+        users = [n.value for n in identity.provides if n.method == "id"]
+        return Q(
+            "terms",
+            **{"access.owned_by.user": users}
+            )
 
 
 class CommunityPermissionPolicy(BasePermissionPolicy):
     """Permissions for Community CRUD operations."""
 
-    can_create = [AnyUser()]
-
-    can_read = [AnyUser()]
-
-    can_update = [AnyUser()]
-
-    can_delete = [AnyUser()]
-
-    can_search = [AnyUser()]
-
-    # can_read_community = [
-    #     CommunityVisibilityPolicy(),
-    # ]
-
-
-    # can_update_community = [
-    #     CommunityMember(['admin', 'curator']),
-    # ]
-    # can_delete_community = [
-    #     CommunityMember(['admin']),
-    # ]
-
-
-def allow_logged_in(*args, **kwargs):
-    """Permission that allows only logged in users to perform the operation."""
-    def can(self):
-        """."""
-        return current_user.is_authenticated
-    return type('AllowCommunityOwner', (), {'can': can})()
-
+    can_create = [AuthenticatedUser()]
 
-# def can_read_community(record, *args, **kwargs):
-#     """Permission wrapper."""
-#     def can(self):
-#         """."""
-#         return CommunityPermissionPolicy(
-#             action='read_community',
-#             comid=record.pid,
-#             community=record,
-#         ).can()
-#     return type('AllowCommunityOwner', (), {'can': can})()
+    can_read = [AnyUserIfPublic()]
 
+    can_update = [CommunityOwner()]
 
-def can_update_community(record, *args, **kwargs):
-    """Permission wrapper."""
-    def can(self):
-        """."""
-        return CommunityPermissionPolicy(
-            action='update_community',
-            comid=record.pid,
-            community=record,
-        ).can()
-    return type('AllowCommunityOwner', (), {'can': can})()
+    can_delete = [CommunityOwner()]
 
+    can_search = [AnyUserIfPublic()]
 
-def can_delete_community(record, *args, **kwargs):
-    """Permission wrapper."""
-    def can(self):
-        """."""
-        return CommunityPermissionPolicy(
-            action='delete_community',
-            comid=record.pid,
-            community=record,
-        ).can()
-    return type('AllowCommunityOwner', (), {'can': can})()
+    can_search_user_communities = [CommunityOwner()]
```

Comment:
This is a bit problematic for protecting the `/api/user/communities` which just requires an `AuthenticatedUser`

Reply:
this permission is only used for the elastic search query while the API validation uses the ```can_search``` one